kratos/includes/data_communicator.h

```diff
@@ -0,0 +1,360 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:     Jordi Cotela
+//
+
+#ifndef KRATOS_DATA_COMMUNICATOR_H_INCLUDED
+#define KRATOS_DATA_COMMUNICATOR_H_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_components.h"
+
+namespace Kratos
+{
+///@addtogroup Kratos Core
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Serial (do-nothing) version of a wrapper class for MPI communication.
+/** @see MPIDataCommunicator for a working distributed memory implementation.
+  */
+class DataCommunicator
+{
+  public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of DataCommunicator
+    KRATOS_CLASS_POINTER_DEFINITION(DataCommunicator);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    DataCommunicator() {}
+
+    /// Destructor.
+    virtual ~DataCommunicator() {}
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    virtual DataCommunicator::UniquePointer Clone() const
+    {
+        return Kratos::make_unique<DataCommunicator>();
+    }
+
+    virtual void Barrier() const {}
+
+    // Reduce operations
+
+    virtual int Sum(const int rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual double Sum(const double rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual array_1d<double,3> Sum(const array_1d<double,3>& rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual int Min(const int rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual double Min(const double rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual array_1d<double,3> Min(const array_1d<double,3>& rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual int Max(const int rLocalValue, const int Root) const
+    {
+        return rLocalValue;
+    }
+
+    virtual double Max(const double rLocalValue, const int Root) const
```

Comment:
it would be good to also have a version of this functions that accepts an std::vector
This can reduce significantly the communication-time in mpi if several values are to be summed etc
Otherwise we have to do [things like this](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FSIapplication/custom_utilities/partitioned_fsi_utilities.hpp#L311-L320)